clients/java/src/main/java/io/camunda/zeebe/client/impl/command/ThrowErrorCommandImpl.java

```diff
@@ -30,69 +33,127 @@
 import java.time.Duration;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Predicate;
+import org.apache.hc.client5.http.config.RequestConfig;
 
 public final class ThrowErrorCommandImpl extends CommandWithVariables<ThrowErrorCommandStep2>
     implements ThrowErrorCommandStep1, ThrowErrorCommandStep2 {
 
   private final GatewayStub asyncStub;
-  private final Builder builder;
+  private final Builder grpcRequestObjectBuilder;
   private final Predicate<StatusCode> retryPredicate;
   private Duration requestTimeout;
+  private final HttpClient httpClient;
+  private final RequestConfig.Builder httpRequestConfig;
+  private final JobErrorRequest httpRequestObject;
+  private boolean useRest;
+  private final long jobKey;
+  private final JsonMapper jsonMapper;
 
   public ThrowErrorCommandImpl(
       final GatewayStub asyncStub,
       final JsonMapper jsonMapper,
       final long key,
       final Duration requestTimeout,
-      final Predicate<StatusCode> retryPredicate) {
+      final Predicate<StatusCode> retryPredicate,
+      final HttpClient httpClient,
+      final boolean preferRestOverGrpc) {
     super(jsonMapper);
     this.asyncStub = asyncStub;
     this.requestTimeout = requestTimeout;
     this.retryPredicate = retryPredicate;
-    builder = ThrowErrorRequest.newBuilder();
-    builder.setJobKey(key);
+    grpcRequestObjectBuilder = ThrowErrorRequest.newBuilder();
+    grpcRequestObjectBuilder.setJobKey(key);
+    this.httpClient = httpClient;
+    httpRequestConfig = httpClient.newRequestConfig();
+    httpRequestObject = new JobErrorRequest();
+    useRest = preferRestOverGrpc;
+    jobKey = key;
+    this.jsonMapper = jsonMapper;
```

Comment:
🔧 Same as with the `FailJobCommandImpl`, why do we need to have a second property for the ` jsonMapper` when the base class already provides it through the `objectMapper` property?

Reply:
Good catch 👍